src/commands/buildCommands/configs/serverAppConfig.js

```diff
@@ -0,0 +1,114 @@
+// Require modules
+const path = require('path');
+const nodeExternals = require('webpack-node-externals');
+const webpack = require('webpack');
+const StatsPlugin = require('webpack-stats-plugin').StatsWriterPlugin;
+const { env, paths } = require('mwp-config');
+
+// Build settings
+const prodPlugins = require('./prodPlugins');
+const rules = require('./rules');
+
+/*
+ * Webpack config object determined by passed-in localeCode. The language is
+ * used to resolve the translated message module paths and determine the output
+ * path
+ *
+ * The server app is a module that exports a rendering function that can be
+ * imported by the server and used to render requests to the app route.
+ */
+function getConfig(localeCode) {
+	const publicPath = `${env.properties.publicPathBase}${localeCode}/`;
+
+	const baseWebfontDir = path.resolve(paths.src.server.app, 'assets', 'fonts');
+	const webfontDir =
+		localeCode === 'ru-RU'
+			? path.resolve(baseWebfontDir, localeCode)
+			: baseWebfontDir;
+	const config = {
+		mode: env.properties.isProd ? 'production' : 'development',
+		entry: {
+			'server-app': [paths.src.server.entry],
+		},
+
+		// write a CommonJS module that can be imported into Node server scripts
+		output: {
+			libraryTarget: 'commonjs2',
+			path: path.join(paths.output.server, localeCode),
+			filename: '[name].js',
+			publicPath,
+		},
+
+		devtool: 'eval',
+
+		module: {
+			rules: [
+				rules.file,
+				rules.scssModule,
+				rules.css,
+				rules.js.server,
+				rules.raw
+			],
+		},
+
+		plugins: [
+			/**
+			 * @see https://webpack.js.org/plugins/environment-plugin/
+			 */
+			new webpack.EnvironmentPlugin({
```

Comment:
No changes needed to webpack.EnvironmentPlugin.

Not sure I understand why we need this plugin though.

Reply:
this plugin will replace all references to `process.env.NODE_ENV` in the code with the actual build-time string value of `NODE_ENV`. React relies on `process.env.NODE_ENV` in order to determine whether to include dev warnings and such, and we use it for similar purposes in application code.

Reply:
Thanks for the explanation! I'm going to elaborate on the existing comment so it's more clear.